stellargraph/connector/neo4j/graph.py

```diff
@@ -34,14 +39,29 @@ class Neo4jStellarGraph:
 
     Args:
         graph (py2neo.Graph): a py2neo.Graph connected to a Neo4j graph database.
+        node_label (str, optional): Common label for all nodes in the graph, if such label exists.
+            Providing this is useful if there are any indexes created on this label (e.g. on node IDs),
+            as it will improve performance of queries.
         is_directed (bool, optional): If True, the data represents a
             directed multigraph, otherwise an undirected multigraph.
     """
 
-    def __init__(self, graph_db, is_directed=False):
+    def __init__(self, graph_db, node_label=None, is_directed=False):
 
         self.graph_db = graph_db
+        self.node_label = node_label
         self._is_directed = is_directed
+        self._node_feature_size = None
+        self._nodes = None
+
+        # FIXME: methods in this class currently only support homogeneous graphs with default node type
+        self._node_type = globalvar.NODE_TYPE_DEFAULT
+
+    def _match_node(self):
+        if self.node_label is not None:
+            return f"MATCH (node:{self.node_label})"
```

Comment:
I think this is vulnerable to a Cypher injection attack, e.g. if someone sets `node_label = "x) ..."` they can run arbitrary Cypher in the `...`. (or something like that). py2neo offers https://py2neo.org/v4/cypher/index.html#py2neo.cypher.cypher_escape which we could use either on a case-by-case basis like here:

```suggestion
            return f"MATCH (node:{cypher_escape(self.node_label)})"
```

Or by storing the labelled escaped. For either case, it might be good to do some basic "Hungarian notation", e.g. have the raw one be `self.raw_node_label` and (if we store it) have the escaped one be `self.cypher_node_label` or `self.escaped_node_label` or something.

Thoughts?

Reply:
Makes sense, although this means we need to import `py2neo` here, so we should move `py2neo` from extra requires to just our regular list of requirements?

Reply:
I think it'd be better to avoid that, so people who aren't using neo4j don't need the dependencies. This also makes it easier for us to be sure that we're not accidentally trying to connect to Neo4j (e.g. on CI). I think it's fine to do an `import py2neo` scoped within this function (or `__init__`) similar to how `Neo4jBatchedNodeGenerator` handles it.

Potentially we should have a `def _py2neo(): ...` function in `neo4j/__init__.py` that manages the import and the error messages for us?

Reply:
The import in `Neo4jBatchedNodeGenerator`'s been removed as part of this PR (don't need it anymore since it's handled in `Neo4jStellarGraph`), so I've just done the same thing in-line without putting it in `neo4j/__init__.py` - I guess we can move it as we need?